scripts/test_upload_dataset.py

```diff
@@ -0,0 +1,50 @@
+from os import name
+
+import pytest
+from azure.cognitiveservices.vision.customvision.training.models import (
+    Project,
+    ProjectSettings,
+    Tag,
+)
+
+from upload_dataset import UploadDataset
+
+upload_dataset = UploadDataset(files_to_upload=[], project_name="test")
+
+
+@pytest.mark.parametrize(
+    "image_path, max_byte_size",
+    [
+        ("/home/oysand/git/eq_robot_car_seal_detection/test_dataset/4.JPG", 4000000),
+        ("/home/oysand/git/eq_robot_car_seal_detection/test_dataset/2102.JPG", 4000000),
+        ("/home/oysand/git/eq_robot_car_seal_detection/test_dataset/2335.JPG", 4000000),
+    ],
+)
+def test_read_and_resize_image(image_path, max_byte_size):
+    image_bytes = upload_dataset._read_and_resize_image(image_path=image_path)
+    assert len(image_bytes) < max_byte_size
+
+
+@pytest.mark.parametrize(
+    "annotation_path",
+    [
+        "/home/oysand/git/eq_robot_car_seal_detection/test_dataset/1.txt",
+        "/home/oysand/git/eq_robot_car_seal_detection/test_dataset/4.txt",
+        "/home/oysand/git/eq_robot_car_seal_detection/test_dataset/34.txt",
+        "/home/oysand/git/eq_robot_car_seal_detection/test_dataset/459.txt",
+        "/home/oysand/git/eq_robot_car_seal_detection/test_dataset/580.txt",
+        "/home/oysand/git/eq_robot_car_seal_detection/test_dataset/611.txt",
+        "/home/oysand/git/eq_robot_car_seal_detection/test_dataset/1189.txt",
+    ],
+)
+def test_read_annotation_file(annotation_path):
+    upload_dataset.green_car_seal_tag = Tag(
+        name="green_car_seal", description="Tag for test", type="For test"
+    )
+    upload_dataset.red_car_seal_tag = Tag(
+        name="red_car_seal", description="Tag for test", type="For test"
+    )
+    annotations: list = upload_dataset._read_annotation_file(
+        annotation_path=annotation_path,
+    )
+    assert type(annotations) == list
```

Comment:
Suggest testing more than just type of annotations.

Reply:
Agreed